include/RAJA/config.hpp.in

```diff
@@ -178,6 +187,25 @@ const int DATA_ALIGN = @RAJA_DATA_ALIGN@;
 #define RAJA_RESTRICT __restrict__
 #endif
 
+//
+// Runtime bounds checking for Views
+//
+#if defined(ENABLE_BOUNDS_CHECK)
+#define BOUNDS_CHECK
+#define BOUNDS_CHECK_constexpr
+
+#if !defined(NDEBUG)
+#define RAJA_ASSERT(EXP) assert( (EXP) )
+#endif
+
+#if defined(NDEBUG)
+#define RAJA_ASSERT(EXP) asm ("trap;")
```

Comment:
how portable is this? will it work with CUDA/HIP and things like Arm or POWER?

Reply:
I've had success with CUDA and power, but I haven't tried on Arm. I can search around to double check.

Reply:
An option is to use abort (standard C) for anything on the host and asm for cuda device code. For Hip...i'm not too sure...

Reply:
@ajkunen, for AMD gpus we can use 
```
      asm("s_trap 2");
 
aborts the kernel with a message to stdout:
 
### HCC STATUS_CHECK Error: HSA_STATUS_ERROR_EXCEPTION (0x1016) at file:mcwamp_hsa.cpp line:1194
Aborted (core dumped)
```